lambda-buffers-compiler/src/LambdaBuffers/Compiler/ProtoCompat/Types.hs

```diff
@@ -217,6 +219,35 @@ data Module = Module
   }
   deriving stock (Show, Eq, Ord, Generic)
 
+data InferenceErr
+  = UnboundTermErr Text
+  | ImpossibleErr Text
+  | UnificationErr Text
+  | RecursiveSubstitutionErr Text
+  deriving stock (Show, Eq, Ord, Generic)
+
+instance Exception InferenceErr
+
+data KindCheckErr
+  = InconsistentTypeErr TyDef
+  | InferenceFailure TyDef InferenceErr
+  deriving stock (Show, Eq, Ord, Generic)
```

Comment:
You're not attaching any SourceInfos. The whole point of them was so I can report back the exact thing that needs fixing. Please find a way to include SourceInfo.

Reply:
Ok will think how to do it meaningfully.